node-backend-starter-code/public/index.html

```diff
@@ -0,0 +1,330 @@
+           *
+           * @param  string   method     HTTP method
+           * @param  string   apiUrl     API endpoint url
+           * @param  mixed    data       Data to be sent. Can be empty
+           * @param  bool     isDataJSON Whether data param is plain text or in JSON
+           * @param  callback callback   This will be called when a valid result is available.
+           *                             Result will be passed in as a JSON object
+           * @param  object   params     Params that will be passed along to the callback besides the result
+           * @return void
+           */
+          function ajaxCall(method, apiUrl, data, isDataJSON, callback, params)
+          {
+              var request = new XMLHttpRequest();
+
+              request.open(method, apiUrl, true);
+              if (isDataJSON) {
+                  request.setRequestHeader('Content-Type', 'application/json');
+              }
+
+              request.onreadystatechange = function () {
+                  if (request.readyState != 4 || request.status != 200) {
+                      return;
+                  }
+
+                  try {
+                      result = JSON.parse(request.responseText);
+                  } catch (exception) {
+                      alert(apiErrorMsg);
+                      return false;
+                  }
+
+                  return callback(result, params);
+              };
+              request.send(data);
+          }
+
+          /**
+           * Helper function to escape single quotes in strings for use in Javascript
+           *
+           * @param  string text
+           * @return string
+           */
+          function escapeString(text)
+          {
+              return text.replace(/'/, "\\'");
+          }
+
+          /**
+           * Retrieve movie details for movie in current page of search results
+           *
+           * @param  int resultIndex
+           * @return false|object
+           */
+          function getMovie(resultIndex)
+          {
+              if (resultIndex < 0 || resultIndex >= searchResults.length) {
+                  return false;
+              }
+
+              return searchResults[resultIndex];
+          }
+
+          /**
+           * Post-AJAX callback when Favorites button is clicked
+           *
+           * @param  false|jsonObject result
+           * @param  object           params Callback params if any
+           * @return void
+           */
+          function favoritesCallback(result, params)
+          {
+              if (0 == result.length) {
+                  favoritesBox.innerHTML = 'You have not saved any movies to your favorites list yet.';
+                  return;
+              }
+
+              var html = '';
+
+              // List items
+              html += '<ol>';
+              result.forEach(function (element, index, array) {
+                  html += '<li>' + element.name + '</li>';
```

Comment:
Correct. Even though the data is coming from a trusted source, it should still be escaped.
Besides escaping element.name, server.js should also validate the data coming in before storing it.